ansible_runner/receptor_plugin.py

```diff
@@ -0,0 +1,174 @@
+import logging
+import json
+import os
+import time
+import io
+import zipfile
+import tempfile
+import uuid
+import asyncio
+
+import ansible_runner.interface
+
+try:
+    import receptor
+    from receptor.config import ReceptorConfig
+    from receptor.controller import Controller
+    receptor_import = True
+except ImportError:
+    receptor_import = False
+
+logger = logging.getLogger(__name__)
+
+
+class UUIDEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, uuid.UUID):
+            return obj.hex
+        return json.JSONEncoder.default(self, obj)
+
+
+# List of kwargs options to the run method that should be sent to the remote executor.
+remote_run_options = (
+    'forks',
+    'host_pattern',
+    'ident',
+    'ignore_logging',
+    'inventory',
+    'limit',
+    'module',
+    'module_args',
+    'omit_event_data',
+    'only_failed_event_data',
+    'playbook',
+    'verbosity',
+)
+
+
+def run_via_receptor(via_receptor, receptor_peer, receptor_node_id, run_options):
+
+    async def read_responses():
+        event_handler = run_options.get('event_handler', None)
+        status_handler = run_options.get('status_handler', None)
+        while True:
+            message = await controller.recv()
+            if message.header.get("eof", False):
+                break
+            elif message.payload:
+                content = json.loads(message.payload.readall())
+                c_header = content[0]
+                c_type = c_header['type']
+                if c_type == 'event':
+                    data = content[1]
+                    if event_handler:
+                        event_handler(data)
+                    if 'stdout' in data and data['stdout']:
```

Comment:
Only if the suppress ansible output is false?

Reply:
The suppress ansible output option is only on the runner config object - it isn't a parameter to the run() method.  Since the present Receptor implementation only supports the run() method, it can't do anything with this.  (See my other comment.)